### ISSUE TITLE
⌨ Support form subject component uses unreliable `MultiSelectFieldBase` component

### ISSUE THREAD

**Issue by mnigh** [GCTC-NTGC/gc-digital-talent#7355]
## ⌨ Accessibility Issue

AT users can't reliably use the component `MultiSelectFieldBase` so it would be beneficial to avoid using it outside of admin. The only implementation outside of admin is in the support form.

REF: https://github.com/GCTC-NTGC/gc-digital-talent/pull/7314#discussion_r1270656926

## 🦋 Expected Behaviour

Field should be able to be used by AT users reliably.

## 📸 Screenshot

![Screen Shot 2023-07-21 at 09 25 23](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/27c23ac2-d9a9-4b51-ac6e-30be06eba997)

## 🙋‍♀️ Proposed Solution

Change `MultiSelectFieldBase` to `Select` component

https://github.com/GCTC-NTGC/gc-digital-talent/blob/d244993542fbe7aaef8d9d38ff6212345e02e8e7/apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx#L173-L213

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] support form subject field uses `Select` component
